### ISSUE TITLE
Convert Gradle config parameters (such as jib.container.labels) to use lazy evaluation

### ISSUE THREAD

**Issue by remy-tiitre** [GoogleContainerTools/jib#3094]
Just like there was an issue to make the jib.to.tags and jib.to.auth to be lazyly evaluated I would need to have the container.labels do the same. Even the reason is the same. I use com.gorylenko.gradle-git-properties Gradle plugin and would like to reuse its properties to add labels to my images. Something like that:
```
jib {
    container {
        labels = [
            'org.label-schema.vcs-url': project.ext.gitProps['git.remote.origin.url'],
            'org.label-schema.vcs-branch': project.ext.gitProps['git.branch'],
            'org.label-schema.vcs-ref': project.ext.gitProps['git.commit.id.abbrev']
        ]
    }
}
```
In our case I would make that configuration global and add this snipped to our custom gradle wrapper init script.

I did try something like this:
```
jib
    container {
        labels = [
            ...
            'org.label-schema.vcs-ref': "${-> project.ext.gitProps['git.commit.id.abbrev']}"
        ]
    }
}
```
But this errored out with this message:
`> class org.codehaus.groovy.runtime.GStringImpl cannot be cast to class java.lang.String (org.codehaus.groovy.runtime.GStringImpl is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @47c62251; java.lang.String is in module java.base of loader 'bootstrap')`

**Comment by chanseokoh** [comment]
Thanks for your feedback! Given the use case, I think making labels (among other parameters) lazily configurable will be very useful.

We also welcome contributions.

Technical notes:
- Going forward, we decided to expose `Property<?>` in [this particular way](https://github.com/GoogleContainerTools/jib/pull/3034#discussion_r581438469) for new parameters. And perhaps this is not possible with existing parameters for compatibility reasons. (Could be revamped in a major version change.)
- Previously, we simply [added a setter for `Property<?>`](https://github.com/GoogleContainerTools/jib/pull/2906/files) for existing parameters.

**Comment by chanseokoh** [comment]
As an immediate workaround, I wonder if setting `jib.container.labels` in a `project.afterEvalaute` block can work for the `com.gorylenko.gradle-git-properties` plugin, like [in this way](https://stackoverflow.com/a/59631660/1701388).

**Comment by NaitYoussef** [comment]
Hello would like to help with this issue

**Comment by chanseokoh** [comment]
@NaitYoussef thanks for your interest!

As noted in https://github.com/GoogleContainerTools/jib/issues/3094#issuecomment-788064416,

> * Going forward, we decided to expose `Property<?>` in [this particular way](https://github.com/GoogleContainerTools/jib/pull/3034#discussion_r581438469) for new parameters. And perhaps this is not possible with existing parameters for compatibility reasons. (Could be revamped in a major version change.)

the way we decided to expose Gradle properties is to use `Property<?>` [like this](https://github.com/GoogleContainerTools/jib/pull/3034/files#diff-ec1f44085899ade105be7cc3c4febe441db808c45bd067be6827bd19c2f52958R238-R244
).

This issue is about `jib.container.labels`, so my guess is that [this property `labels`](https://github.com/GoogleContainerTools/jib/blob/90359337e136800b4ec4e6abfc5b6816c2fa6a4e/jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ContainerParameters.java#L46) should be changed to be of type [`MapProperty`](https://docs.gradle.org/current/javadoc/org/gradle/api/provider/MapProperty.html). This may not be fully backward-compatible, but I think should be fine most of the time.

**Comment by chanseokoh** [comment]
Fixed by #3094.

**Comment by chanseokoh** [comment]
@remy-tiitre @ChristianCiach @NaitYoussef Jib 3.1.0 is released with this support. Let us know if it doesn't work.

**Comment by jamesdh** [comment]
@chanseokoh I believe you meant Fixed by #3242? 

It might be worth mentioning somewhere that this can cause a breakage if using a Kotlin based build.gradle, causing a `Val cannot be reassigned` error.

**Comment by chanseokoh** [comment]
@jamesdh I guess you'll have to use `.set()` in Kotlin? Is that the only possible way?
